oidc/client.py

```diff
@@ -105,6 +106,93 @@ def _retrieve_jwks_related_to_kid(self, kid: str) -> Optional[str]:
                     return jwk
         return None
 
+    def retrieve_user_info(self, access_token: str) -> dict:
+        """
+        Given an access_token, retrieve user profile claims
+        """
+        response = requests.get(
+            self.user_info_endpoint, headers={"Authorization": f"Bearer {access_token}"}
+        )
+        return response.json()
+
+    def get_hash_algorithm(self, alg: str) -> Optional[Callable]:
+        """
+        Maps JWT algorithm to hash function.
+
+        Based on the spec: RS256/ES256/PS256 use SHA-256,
+        RS384/ES384/PS384 use SHA-384, RS512/ES512/PS512 use SHA-512
+        """
+        algorithm_map = {
+            "RS256": hashlib.sha256,
+            "RS384": hashlib.sha384,
+            "RS512": hashlib.sha512,
+            "ES256": hashlib.sha256,
+            "ES384": hashlib.sha384,
+            "ES512": hashlib.sha512,
+            "PS256": hashlib.sha256,
+            "PS384": hashlib.sha384,
+            "PS512": hashlib.sha512,
+            "HS256": hashlib.sha256,
+            "HS384": hashlib.sha384,
+            "HS512": hashlib.sha512,
+        }
+        return algorithm_map.get(alg)
+
+    def validate_access_token(
+        self, verifyied_id_token: dict, id_token: str, access_token: str
+    ) -> bool:
+        """
+        Validates an access token against the at_hash claim in an ID token.
+
+        Args:
```

Comment:
Let's stick to the [restructured text docstring format](https://realpython.com/how-to-write-docstrings-in-python/#restructuredtext-docstrings)